gdal/ogr/ogrsf_frmts/cad/gdalcaddataset.cpp

```diff
@@ -0,0 +1,463 @@
+            CPLString osImgFilename = pImage->getFilePath();
+            CPLString osImgPath = CPLGetPath(osImgFilename);
+            if(osImgPath.empty ())
+            {
+                osImgFilename = CPLFormFilename(CPLGetPath(osCADFilename),
+                                                osImgFilename, NULL);
+            }
+
+            if ( CPLCheckForFile ((char*)osImgFilename.c_str (), NULL) == FALSE)
+                return poOpenInfo->nOpenFlags & GDAL_OF_VECTOR;
+
+            poRasterDS = reinterpret_cast<GDALDataset *>(
+                                GDALOpen( osImgFilename, poOpenInfo->eAccess ) );
+            if(poRasterDS == NULL)
+            {
+                delete pImage;
+                return poOpenInfo->nOpenFlags & GDAL_OF_VECTOR;
+            }
+            if(poRasterDS->GetRasterCount() == 0)
+            {
+                delete pImage;
+                GDALClose( poRasterDS );
+                return poOpenInfo->nOpenFlags & GDAL_OF_VECTOR;
+            }
+
+            if(poRasterDS->GetGeoTransform(adfGeoTransform) != CE_None)
+            {
+                // external world file have priority
+                double dfUnits = 1;
+                if(NULL != poSpatialRef)
+                    dfUnits = poSpatialRef->GetLinearUnits();
+                FillTransform(pImage, dfUnits);
+            }
+            delete pImage;
+
+            nRasterXSize = poRasterDS->GetRasterXSize();
+            nRasterYSize = poRasterDS->GetRasterYSize();
+            if (!GDALCheckDatasetDimensions(nRasterXSize, nRasterYSize))
+            {
+                GDALClose( poRasterDS );
+                return poOpenInfo->nOpenFlags & GDAL_OF_VECTOR;
+            }
+
+            for( int iBand = 1; iBand <= poRasterDS->GetRasterCount(); iBand++ )
+                SetBand( iBand,
+                    new CADWrapperRasterBand( poRasterDS->GetRasterBand( 
+                        iBand )) );
+
+            char** papszDomainList = poRasterDS->GetMetadataDomainList();
+            while( papszDomainList )
+            {
+                char** papszMetadata = GetMetadata(*papszDomainList);
+                char** papszRasterMetadata = poRasterDS->GetMetadata(*papszDomainList);
+                if(NULL == papszMetadata)
+                    SetMetadata(papszRasterMetadata, *papszDomainList);
+                else
+                    papszMetadata = CSLMerge(papszMetadata, papszRasterMetadata);    
+                papszDomainList++;
+            }
+        }
+    }
+
+    return TRUE;
+}
+
+OGRLayer *GDALCADDataset::GetLayer( int iLayer )
+{
+    if ( iLayer < 0 || iLayer >= nLayers )
+        return( NULL );
+    else
+        return( papoLayers[iLayer] );
+}
+
+int GDALCADDataset::TestCapability( const char * pszCap )
+{
+    if ( EQUAL(pszCap,ODsCCreateLayer) ||
+         EQUAL(pszCap,ODsCDeleteLayer) )
+    {
+         return FALSE;
+    }
+    else if( EQUAL(pszCap,ODsCCurveGeometries) )
+        return TRUE;
```

Comment:
If you have a {} scope for one par of if else, you should have it for the rest.  Consistency reduces chances of later mistakes during maintenance.